capstone-rs/src/capstone.rs

```diff
@@ -381,7 +381,7 @@ impl Capstone {
     /// 1. Instruction was created with detail enabled
     /// 2. Skipdata is disabled
     /// 3. Capstone was not compiled in diet mode
-    pub fn insn_detail<'s, 'i: 's>(&'s self, insn: &'i Insn) -> CsResult<InsnDetail<'i>> {
+    pub fn insn_detail<'s, 'i: 's>(&'s self, insn: &'i impl Instruction) -> CsResult<InsnDetail<'i>> {
```

Comment:
It may be possible to still be using `Insn<'_>` here, and just have `OwnedInsn<'captstone>` deref to `Insn<'capstone>` through a pointer cast. That ought to alleviate the resulting ergonomics / API readability quite a bit 🙂